src/AzureOpenAIProxy.Management/Components/Pages/ModelDelete.razor.cs

```diff
@@ -0,0 +1,76 @@
+using AzureOpenAIProxy.Management.Components.ModelManagement;
+using AzureOpenAIProxy.Management.Database;
+using Microsoft.AspNetCore.Components;
+using Microsoft.EntityFrameworkCore;
+
+namespace AzureOpenAIProxy.Management.Components.Pages;
+
+public partial class ModelDelete : ComponentBase
+{
+    [Parameter]
+    public required string Id { get; set; }
+
+    [Inject]
+    public AoaiProxyContext DbContext { get; set; } = null!;
+
+    [Inject]
+    public NavigationManager NavigationManager { get; set; } = null!;
+
+    public ModelEditorModel Model { get; set; } = null!;
+
+    protected override async Task OnInitializedAsync()
+    {
+        if (string.IsNullOrEmpty(Id))
+        {
+            NavigationManager.NavigateTo("/models");
+            return;
+        }
+        OwnerCatalog? m = await DbContext.OwnerCatalogs.FindAsync(Guid.Parse(Id));
+
+        if (m is null)
+        {
+            NavigationManager.NavigateTo("/models");
+            return;
+        }
+
+        Model = new()
+        {
+            FriendlyName = m.FriendlyName,
+            DeploymentName = m.DeploymentName,
+            EndpointKey = m.EndpointKey,
+            ModelType = m.ModelType,
+            EndpointUrl = m.EndpointUrl,
+            Location = m.Location,
+            Active = m.Active,
+        };
+    }
+
+    private async Task OnValidSubmit(ModelEditorModel model)
+    {
+        OwnerCatalog? m = await DbContext.OwnerCatalogs.FindAsync(Guid.Parse(Id));
+
+        if (m is null)
+        {
+            NavigationManager.NavigateTo("/models");
+            return;
+        }
+
+        // check if the catalogId is in EventCatalogMap
+        var eventMap = await DbContext.Events.FirstOrDefaultAsync(x => x.Catalogs.Any(c => c.CatalogId == m.CatalogId));
+
+        // check if the catalogId is in Metric
+        var metric = await DbContext.Metrics.FirstOrDefaultAsync(x => x.CatalogId == m.CatalogId);
```

Comment:
Shouldn't this be solved via cascading deletes, or did we remove relationships between the types?

Reply:
Or is the goal to block deletes if there are metrics and/or events?

Reply:
> Shouldn't this be solved via cascading deletes, or did we remove relationships between the types?

I think we removed as we wanted historic record. But 100% sure that is the right decision...

Reply:
> Or is the goal to block deletes if there are metrics and/or events?

That is what we decided. We want to keep historic data